guides/testing.md

```diff
@@ -0,0 +1,163 @@
+# Testing Crystal Code
+
+Crystal comes with a full-feature spec library in the [`Spec` module](https://crystal-lang.org/api/latest/Spec.html). It provides the structure for writing executable examples of how your code should behave.
+
+Inspired by [Rspec](http://rspec.info/) it includes a domain specific language (DSL) that allows to write examples in a way similar to plain english..
+
+A basic spec looks something like this:
+
+```crystal
+require "spec"
+
+describe Array do
+  describe "#size" do
+    it "correctly reports the number of elements in the Array" do
+      [1, 2, 3].size.should eq 3
+    end
+  end
+
+  describe "#empty?" do
+    it "is empty when no elements are in the array" do
```

Comment:
I'd use "is true" and "is false" instead of "is empty" and "is not empty" because we are describing a method, not array. This means it reads "Array #empty? is true when no elements are in the array" instead of "Array #empty? is empty when no elements are in the array".